flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/expressions/RexNodeConverter.java

```diff
@@ -529,7 +530,8 @@ private RexNode convertOver(List<Expression> children) {
 				.collect(Collectors.toList());
 		// assemble bounds
 		Expression preceding = args.get(2);
-		boolean isPhysical = ((ResolvedExpression) preceding).getOutputDataType().equals(DataTypes.BIGINT());
+		boolean isPhysical = fromDataTypeToTypeInfo(((ResolvedExpression) preceding).getOutputDataType()).equals(
```

Comment:
what's the difference between LongTypeInfo and BIGINT datatype?

Reply:
TypeInformation has no nullable information while DataType has.
There is no need to compare nullable here.

Reply:
How about we ignore the nullability during comparison, but not convert to TypeInformation. It seems unnecessary.

Reply:
Here, I think we need use `LogicalTypeRoot`, because we don't want to judge `nullable` and `conversionClass`.

Reply:
Ok